### DISCUSSION TITLE
Sorting removed on third click on header

### DISCUSSION THREAD

**Discussion by natashajaved** [TanStack/table#2848] [Q&A]
I am using react-table v7. I have added sorted to my table though sorting has this behavior that when a header is clicked the third time the sorting is removed. I want the sorting to be toggled once sorting is applied to a header e.g. If I clicked header and now it's sorted as ascending then on second click to descending then on third click back to descending and so on. Sorting should only be removed when some other header is clicked.

Here is the code snippet
`import React from 'react'
import { useSortBy, useTable } from 'react-table'

const Grid = (props) => {
    console.log(props)
    const { data, columns, showFooter = false, disableSort = false } = props
    const {
        getTableProps,
        getTableBodyProps,
        headerGroups,
        rows,
        prepareRow,
        footerGroups
    } = useTable({ columns, data, disableSortBy: false }, useSortBy)
    return (
        <div className="radius_16 bg-white p-3 mt-3">
            <div className="table_border">
                <table {...getTableProps()} className="table tr_bg mb-0">
                    <thead>
                        {headerGroups.map(headerGroup => (
                            <tr {...headerGroup.getHeaderGroupProps()}>
                                {headerGroup.headers.map(column => {
                                    console.log(column)
                                    return <th
                                        {...column.getHeaderProps(column.getSortByToggleProps())}
                                    >
                                        <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                                            {column.render('Header')}
                                            {<i className={`fad color_orange cursor_pointer sorting  ${column.isSorted ? column.isSortedDesc ? 'fa-sort-down' : 'fa-sort-up' : ''}`} style={{ minWidth: 10 }} />}
                                        </div>
                                    </th>
                                })}
                            </tr>
                        ))}
                    </thead>
                    <tbody {...getTableBodyProps()}>
                        {rows.map(row => {
                            prepareRow(row)
                            return (
                                <tr {...row.getRowProps()}>
                                    {row.cells.map(cell => {
                                        return (
                                            <td
                                                {...cell.getCellProps()}
                                            >
                                                {cell.render('Cell')}
                                            </td>
                                        )
                                    })}
                                </tr>
                            )
                        })}
                    </tbody>
                    {showFooter && <tfoot>
                        {footerGroups.map(group => (
                            <tr {...group.getFooterGroupProps()} className="bg_dark_gray">
                                {group.headers.map(column => (
                                    <td {...column.getFooterProps()}>{column.render('Footer')}</td>
                                ))}
                            </tr>
                        ))}
                    </tfoot>}
                </table>
            </div>
        </div>
    )
}`

**Comment by wildcat-cs** [comment]
I think this you are looking for disableSortRemove :

```
const {
...
} = useTable({ columns, data, disableSortRemove:true }, useSortBy)
```

**Reply by natashajaved** [reply]
@wildcat-cs YES! Thanks, this is what I was looking for.

**Reply by vittxr** [reply]
In v8, the api change the attribute name to `enableSortingRemoval` and you need to set it to false to have the same behaviour
